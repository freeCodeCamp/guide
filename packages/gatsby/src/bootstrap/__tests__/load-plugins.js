const loadPlugins = require(`../load-plugins`)

describe(`Load plugins`, () => {
  it(`load plugins for a site`, async () => {
    let plugins = await loadPlugins({ plugins: [] })

    // Delete the resolve path as that varies depending
    // on platform so will cause snapshots to differ.
    plugins = plugins.map(plugin => {
      return {
        ...plugin,
        resolve: ``,
      }
    })

    expect(plugins).toMatchSnapshot()
  })

  it(`Loads plugins defined with an object but without an option key`, async () => {
    const config = {
      plugins: [
        {
          resolve: `___TEST___`,
        },
      ],
    }

    let plugins = await loadPlugins(config)

    // Delete the resolve path as that varies depending
    // on platform so will cause snapshots to differ.
    plugins = plugins.map(plugin => {
      return {
        ...plugin,
        resolve: ``,
      }
    })

    expect(plugins).toMatchSnapshot()
  })
})
