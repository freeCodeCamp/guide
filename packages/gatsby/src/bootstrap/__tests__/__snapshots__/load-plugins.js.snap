// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Load plugins Loads plugins defined with an object but without an option key 1`] = `
Array [
  Object {
    "name": "component-page-creator",
    "pluginOptions": Object {
      "plugins": Array [],
    },
    "resolve": "",
    "version": "1.0.0",
  },
  Object {
    "name": "internal-data-bridge",
    "pluginOptions": Object {
      "plugins": Array [],
    },
    "resolve": "",
    "version": "1.0.0",
  },
  Object {
    "name": "dev-404-page",
    "pluginOptions": Object {
      "plugins": Array [],
    },
    "resolve": "",
    "version": "1.0.0",
  },
  Object {
    "name": "query-runner",
    "pluginOptions": Object {
      "plugins": Array [],
    },
    "resolve": "",
    "version": "1.0.0",
  },
  Object {
    "name": "TEST",
    "pluginOptions": Object {
      "plugins": Array [],
    },
    "resolve": "",
    "version": undefined,
  },
  Object {
    "name": "default-site-plugin",
    "pluginOptions": Object {
      "plugins": Array [],
    },
    "resolve": "",
    "version": "n/a",
  },
]
`;

exports[`Load plugins load plugins for a site 1`] = `
Array [
  Object {
    "name": "component-page-creator",
    "pluginOptions": Object {
      "plugins": Array [],
    },
    "resolve": "",
    "version": "1.0.0",
  },
  Object {
    "name": "internal-data-bridge",
    "pluginOptions": Object {
      "plugins": Array [],
    },
    "resolve": "",
    "version": "1.0.0",
  },
  Object {
    "name": "dev-404-page",
    "pluginOptions": Object {
      "plugins": Array [],
    },
    "resolve": "",
    "version": "1.0.0",
  },
  Object {
    "name": "query-runner",
    "pluginOptions": Object {
      "plugins": Array [],
    },
    "resolve": "",
    "version": "1.0.0",
  },
  Object {
    "name": "default-site-plugin",
    "pluginOptions": Object {
      "plugins": Array [],
    },
    "resolve": "",
    "version": "n/a",
  },
]
`;
